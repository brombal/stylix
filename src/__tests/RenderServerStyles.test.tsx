import { render } from '@testing-library/react';
import $, { RenderServerStyles, StylixProvider } from '../index';

describe('RenderServerStyles', () => {
  it('should work with basic styles', () => {
    const result = render(
      <StylixProvider ssr>
        <$.div color="red">red</$.div>
        <$.div color="green">green</$.div>
        <RenderServerStyles />
      </StylixProvider>,
    );

    expect(result.container.innerHTML).toMatchSnapshot();
  });

  it('should not render anything on client', () => {
    const result = render(
      <StylixProvider ssr={false}>
        <$.div color="red">red</$.div>
        <$.div color="green">green</$.div>
        <RenderServerStyles />
      </StylixProvider>,
    );

    expect(result.container.innerHTML).toMatchSnapshot();
  });
});
