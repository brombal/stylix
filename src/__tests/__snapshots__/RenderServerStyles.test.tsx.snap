// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderServerStyles should not render anything on client 1`] = `"<div class="stylix-1">red</div><div class="stylix-2">green</div><style type="text/css"></style>"`;

exports[`RenderServerStyles should work with basic styles 1`] = `
"<div class="stylix-1">red</div><div class="stylix-2">green</div><style type="text/css">.stylix-1 {
  color: red; }
.stylix-2 {
  color: green; }</style>"
`;
